### PR TITLE
fix snapshot diffs for serialization changes

### PR DESCRIPTION
**Pull request by evanmcc**
also do more work to filter out encoding-related non-changes.

### CODE REVIEW

File: src/blockchain.erl [hunk 1]
```diff
@@ -1438,6 +1433,42 @@ crosscheck(Blockchain) ->
             Error
     end.
 
+compare(LedgerA, LedgerB) ->
```
Comment (vihu) [inline src/blockchain.erl:1436]:
```suggestion
-spec compare(LedgerA :: blockchain_ledger_v1:ledger(), LedgerB :: blockchain_ledger_v1:ledger()) -> ok | {error, any()}.
compare(LedgerA, LedgerB) ->
```

Comment (vihu) [reply src/blockchain.erl:1436]: also wondering if `compare/2` should be put in ledger module..

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 2]
```diff
@@ -590,8 +590,8 @@ raw_fingerprint(#ledger_v1{mode = Mode} = Ledger, Extended) ->
             false ->
                 {ok, #{<<"ledger_fingerprint">> => LedgerHash}};
             _ ->
-                [GWsHash, EntriesHash, DCEntriesHash, HTLCsHash,
-                 PoCsHash, SecuritiesHash, RoutingsHash, StateChannelsHash, SubnetsHash, _] = L,
+                [_, GWsHash, EntriesHash, DCEntriesHash, HTLCsHash,
```
Comment (vihu) [inline src/ledger/v1/blockchain_ledger_v1.erl:593]: Not sure I understand why we ignore `DefaultHash` here, isn't the same we put in the `core_fingerprint` value every time we calculated above?

Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_v1.erl:593]: all of these are calculated about, we're just matching out all the hashes here to have some bindings to use.  `DefaultHash` is already bound, so why force a match calculation when we rebind it?

Comment (vihu) [reply src/ledger/v1/blockchain_ledger_v1.erl:593]: makes sense, thanks

File: src/ledger/v1/blockchain_ledger_gateway_v2.erl [hunk 3]
```diff
@@ -328,60 +328,60 @@ print(Address, Gateway, Ledger, Verbose) ->
...
-        {ok, Witness=#witness{nonce=Nonce, count=Count, hist=Hist}} ->
+    case lists:keytake(WitnessAddress, 1, Witnesses) of
+        {value, {_, Witness=#witness{nonce=Nonce, count=Count, hist=Hist}}, Witnesses1} ->
             %% nonce is the same, increment the count
-            Gateway#gateway_v2{witnesses=maps:put(WitnessPubkeyBin,
-                                                  Witness#witness{count=Count + 1,
-                                                                  hist=update_histogram(RSSI, Hist),
-                                                                  recent_time=TS},
-                                                  Witnesses)};
+            Gateway#gateway_v2{witnesses=[{WitnessAddress,
```
Comment (vihu) [inline src/ledger/v1/blockchain_ledger_gateway_v2.erl:341]: the witnesses list here in gateway_v2 does not need to be sorted?

**Pull request merged**